Fix addModlist on python3.

bbaetz opened the pull request:
With python3 filter returns a filter object, which will always be true, so
convert to a list to get the actual entries.

Note that this already has a test (test_modlist.py) that picked this up,
but its not a standard python unittest so isn't run by default.

tiran commented:
How about we replace the line with proper, modern Python code?

old:
```filter(lambda x:x!=None,entry[attrtype])```

new:
```list(item for item in entry[attrtype] if item is not None)```

bbaetz committed:
Fix addModlist on python3.

With python3 filter returns a filter object, which will always be true.
Use a more-pythonic list comprehension instead of filter.

Update other usage of this pattern to not use filter either.

Note that this already has a test (test_modlist.py) that picked this up,
but its not a standard python unittest so isn't run by default.

bbaetz commented:
Done, and updated the other list(filter(..)) users too.
